api/src/main/java/org/openmrs/module/diagnosis/DiagnosisService.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.diagnosis;
+
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.OpenmrsService;
+import org.openmrs.Diagnosis;
+
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * <pre>
+ * API methods for managing diagnoses
+ * </pre>
+ */
+public interface DiagnosisService extends OpenmrsService {
+
+	/**
+	 * Changes a non-coded diagnosis to coded diagnosis
+	 * @param nonCodedObs an Obs that contains the non-coded diagnosis
+	 * @param diagnoses a List of Diagnosis representing the new diagnoses
+	 * @return
+	 */
+	List<Obs> codeNonCodedDiagnosis(Obs nonCodedObs, List<Diagnosis> diagnoses);
+
+	/**
+	 * Gets diagnoses since date, sorted in reverse chronological order
+	 *
+	 * @param patient
```

Comment:
Can we also document each of these params?

Reply:
Sure